I've checked again, previous error happen because we tried to install Airflow from PyPI with constraints for main branch.
If it failed it tried to install airflow without constraints.

And actual error happen in tests

```
  ==================================== ERRORS ====================================
  _______________ ERROR collecting test_kubernetes_pod_operator.py _______________
  ImportError while importing test module '/home/runner/work/airflow/airflow/kubernetes_tests/test_kubernetes_pod_operator.py'.
  Hint: make sure your test modules/packages have valid Python names.
  Traceback:
  /opt/hostedtoolcache/Python/3.8.18/x64/lib/python3.8/importlib/__init__.py:127: in import_module
      return _bootstrap._gcd_import(name[level:], package, level)
  test_kubernetes_pod_operator.py:41: in <module>
      from airflow.providers.cncf.kubernetes.hooks.kubernetes import KubernetesHook
  ../airflow/providers/cncf/kubernetes/hooks/kubernetes.py:25: in <module>
      import aiofiles
  E   ModuleNotFoundError: No module named 'aiofiles'
```

I would recommend to add `aiofiles` into the https://github.com/apache/airflow/blob/main/scripts/ci/kubernetes/k8s_requirements.txt as workaround

Jarek @potiuk maybe you know how to deal with new dependencies for K8S Provider in tests without add it into ci requirements?

--- a/airflow/providers/cncf/kubernetes/hooks/kubernetes.py
+++ b/airflow/providers/cncf/kubernetes/hooks/kubernetes.py
@@ -16,13 +16,13 @@
 # under the License.
 from __future__ import annotations
 
-import aiofiles
 import contextlib
 import json
 import tempfile
 from functools import cached_property
 from typing import TYPE_CHECKING, Any, Generator
 
+import aiofiles
 from asgiref.sync import sync_to_async
 from kubernetes import client, config, watch
 from kubernetes.config import ConfigException

--- a/airflow/providers/cncf/kubernetes/hooks/kubernetes.py
+++ b/airflow/providers/cncf/kubernetes/hooks/kubernetes.py
@@ -503,7 +503,7 @@ class AsyncKubernetesHook(KubernetesHook):
             return async_client.ApiClient()
 
         if kubeconfig is not None:
-            async with aiofiles.tempfile.TemporaryFile("wb") as temp_config:
+            async with aiofiles.tempfile.NamedTemporaryFile() as temp_config:
                 self.log.debug(
                     "Reading kubernetes configuration file from connection "
                     "object and writing temporary config file with its content",

--- a/airflow/providers/cncf/kubernetes/hooks/kubernetes.py
+++ b/airflow/providers/cncf/kubernetes/hooks/kubernetes.py
@@ -22,6 +22,7 @@ import tempfile
 from functools import cached_property
 from typing import TYPE_CHECKING, Any, Generator
 
+import aiofiles
 from asgiref.sync import sync_to_async
 from kubernetes import client, config, watch
 from kubernetes.config import ConfigException
@@ -502,13 +503,13 @@ class AsyncKubernetesHook(KubernetesHook):
             return async_client.ApiClient()
 
         if kubeconfig is not None:
-            with tempfile.NamedTemporaryFile() as temp_config:
+            async with aiofiles.tempfile.TemporaryFile("wb") as temp_config:
                 self.log.debug(
                     "Reading kubernetes configuration file from connection "
                     "object and writing temporary config file with its content",
                 )
-                temp_config.write(kubeconfig.encode())
-                temp_config.flush()
+                await temp_config.write(kubeconfig.encode())
+                await temp_config.flush()
                 self._is_in_cluster = False
                 await async_config.load_kube_config(
                     config_file=temp_config.name,

--- a/airflow/providers/cncf/kubernetes/provider.yaml
+++ b/airflow/providers/cncf/kubernetes/provider.yaml
@@ -69,6 +69,7 @@ versions:
   - 1.0.0
 
 dependencies:
+  - aiofiles>=23.2.0
   - apache-airflow>=2.5.0
   - asgiref>=3.5.2
   - cryptography>=2.0.0

--- a/generated/provider_dependencies.json
+++ b/generated/provider_dependencies.json
@@ -258,6 +258,7 @@
   },
   "cncf.kubernetes": {
     "deps": [
+      "aiofiles>=23.2.0",
       "apache-airflow>=2.5.0",
       "asgiref>=3.5.2",
       "cryptography>=2.0.0",

--- a/setup.py
+++ b/setup.py
@@ -385,6 +385,7 @@ mypy_dependencies = [
     # Make sure to upgrade the mypy version in update-common-sql-api-stubs in .pre-commit-config.yaml
     # when you upgrade it here !!!!
     "mypy==1.2.0",
+    "types-aiofiles",
     "types-certifi",
     "types-croniter",
     "types-Deprecated",

